services/search/pkg/engine/bleve/index.go

```diff
@@ -0,0 +1,150 @@
+package bleve
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/blevesearch/bleve/v2"
+	"github.com/blevesearch/bleve/v2/mapping"
+)
+
+// IndexGetter is an interface that provides a way to get an index.
+// Implementations might differ in how the index is created and how the
+// index is gotten (reused, created on the fly, etc).
+//
+// Some implementations might require the index to be kept opened, meaning
+// the index should be closed only when the application is shutting down. In
+// this case, IndexCanBeClosed should return false. If the index can be
+// closed and reopened safely at any time, IndexCanBeClosed should
+// return true.
+type IndexGetter interface {
+	GetIndex(opts ...GetIndexOption) (bleve.Index, error)
```

Comment:
We could change the interface slightly here. If we return the closer, we get around the `IndexCanBeClosed` method
```suggestion
	GetIndex(opts ...GetIndexOption) (bleve.Index, func(), error)
```
We can then always do 
```go
index, close, err := b.GetIndex()
if err != nil {
    return
}
defer close()
```
This would allow us to skip all the bool checks